File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    cnt = 0
    for i in range(1,n+1):
        tmp = 0
        for j in range(i,n+1):
            tmp += j
            if tmp > n:
                tmp = 0
                break
            elif tmp == n:
                cnt += 1
                tmp = 0
                break
    return cnt

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    for (int i = 1; i <= n;i++)
    {
        int tmp = 0;
        for (int j = i; j <= n; j++)
        {    
            tmp += j;
            if (tmp == n)
            {
                answer++;
                break;
            }
            else if (tmp > n)
                break;                
        }
    }
    return answer;
}